### PR TITLE
feature(locksmith): Using OpenZeppelin's Relay to send transactions and sign messages

### PR DESCRIPTION
**Pull request by julien51**
# Description

This was easier than I thought but probably needs a lot more testing.
Also, right now this is on a _per network_ basis, which is ok I guess, but we need to find a way to avoid having 1 API key/secret pair for each network or this will get hard to manage pretty quickly IMO.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/unlock-js/src/KeyManager.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export interface TransferObject {
   deadline: number
 }
 
-type Signer = ethers.Wallet | ethers.providers.JsonRpcSigner
+type Signer = ethers.Signer | ethers.providers.JsonRpcSigner
```
Comment (julien51) [inline packages/unlock-js/src/KeyManager.ts:16]: Changed to signer to be more accurate!

**Comment by julien51** [comment]
First transaction sent by the Relayer! https://goerli.etherscan.io/tx/0x0e8a00b37373074c50a1ced32c8ec19669d8470bc3b382bb783df300b724dc49

**Comment by julien51** [comment]
Bad news: we need to use differenyt API keys for every network. That's going to be quite a bore :(

**Comment by julien51** [comment]
Using 1password for secrets should work! https://github.com/unlock-protocol/unlock/pull/12162

**Comment by julien51** [comment]
Since we will be adding so many secrets, we need a better secret approach: https://github.com/unlock-protocol/unlock/issues/12002

File: locksmith/src/config/config.ts [hunk 2]
```diff
@@ -58,7 +58,8 @@ const config = {
   recaptchaSecret: process.env.RECAPTCHA_SECRET,
   logtailSourceToken: process.env.LOGTAIL,
   sessionDuration: Number(process.env.SESSION_DURATION || 86400 * 60), // 60 days
-  requestTimeout: '30s',
+  requestTimeout: '25s',
```
Comment (julien51) [inline locksmith/src/config/config.ts:61]:
```suggestion
  requestTimeout: '30s',
```

**Review by searchableguy — APPROVED** [review]

**Pull request merged**